Pcap++/src/PcapLiveDevice.cpp

```diff
@@ -488,41 +499,67 @@ int PcapLiveDevice::startCaptureBlockingMode(OnPacketArrivesStopBlocking onPacke
 	m_cbOnPacketArrivesBlockingMode = onPacketArrives;
 	m_cbOnPacketArrivesBlockingModeUserCookie = userCookie;
 
-	long startTimeSec = 0, startTimeNSec = 0;
-	clockGetTime(startTimeSec, startTimeNSec);
-
-	long curTimeSec = 0;
-
 	m_CaptureThreadStarted = true;
 	m_StopThread = false;
 
-	if (timeout <= 0)
+	const int64_t timeoutMs = timeout * 1000; // timeout unit is seconds, let's change it to milliseconds
+	auto startTime = std::chrono::steady_clock::now();
+	auto currentTime = startTime;
+
+#if !defined(_WIN32)
+	struct pollfd pcapPollFd;
+	memset(&pcapPollFd, 0, sizeof(pcapPollFd));
+	pcapPollFd.fd = m_PcapSelectableFd;
+	pcapPollFd.events = POLLIN;
+#endif
+
+	if(timeoutMs <= 0)
 	{
 		while (!m_StopThread)
 		{
 			pcap_dispatch(m_PcapDescriptor, -1, onPacketArrivesBlockingMode, (uint8_t*)this);
 		}
-		curTimeSec = startTimeSec + timeout;
 	}
 	else
 	{
-		while (!m_StopThread && curTimeSec <= (startTimeSec + timeout))
+		while (!m_StopThread && std::chrono::duration_cast<std::chrono::milliseconds>(currentTime  - startTime).count() <= timeoutMs )
 		{
-			long curTimeNSec = 0;
-			pcap_dispatch(m_PcapDescriptor, -1, onPacketArrivesBlockingMode, (uint8_t*)this);
-			clockGetTime(curTimeSec, curTimeNSec);
+			if(m_usePoll)
+			{
+#if !defined(_WIN32)
+				int64_t pollTimeoutMs = timeoutMs - std::chrono::duration_cast<std::chrono::milliseconds>(currentTime  - startTime).count();
+				pollTimeoutMs = std::max(pollTimeoutMs, (int64_t)0); // poll will be in blocking mode if negative value
+				int ready = poll(&pcapPollFd, 1, pollTimeoutMs); // wait the packets until timeout
+				if(ready > 0)
+				{
+					pcap_dispatch(m_PcapDescriptor, -1, onPacketArrivesBlockingMode, (uint8_t*)this);
+					currentTime = std::chrono::steady_clock::now();
+				}
+				else if(ready < 0)
+				{
+					PCPP_LOG_ERROR("poll() got error '" <<  strerror(errno) << "'");
+					return -1;
+				}
+#endif
```

Comment:
We can add `#else`:
```cpp
#else
				PCPP_LOG_ERROR("Windows doesn't support poll()");
				return 0;
#endif
```

Reply:
should not going to here, but sure, just in case